protostar-cli/src/starknet_commands/call.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::{Error, Result};
+use clap::Args;
+use starknet::core::types::{BlockId, FieldElement, FunctionCall};
+use starknet::core::utils::get_selector_from_name;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::{JsonRpcClient, Provider};
+
+#[derive(Args)]
+#[command(about = "Call a contract instance on Starknet", long_about = None)]
+pub struct Call {
+    /// Address of the called contract (hex)
+    #[clap(short = 'a', long = "contract-address")]
+    pub(crate) contract_address: String,
+
+    /// Name of the contract function to be called
+    #[clap(short = 'e', long = "func-name")]
+    pub(crate) func_name: String,
+
+    /// Arguments of the called function (list of hex)
+    #[clap(short = 'c', long = "calldata", num_args = 1.., value_delimiter = ' ')]
+    pub(crate) calldata: Option<Vec<String>>,
+
+    /// Block identifier on which call should be performed
+    #[clap(short = 'b', long = "block-id", default_value = "pending")]
+    pub(crate) block_id: String,
+}
+
+pub async fn call(
+    contract_address: &str,
+    func_name: &str,
+    calldata: &Option<Vec<String>>,
+    provider: &JsonRpcClient<HttpTransport>,
+    block_id: &BlockId,
+) -> Result<Vec<FieldElement>, Error> {
+    let function_call = FunctionCall {
+        contract_address: FieldElement::from_hex_be(contract_address)?,
+        entry_point_selector: get_selector_from_name(func_name)?,
```

Comment:
You can consider adding additional context to the errors so the resulting error is clearer when displayed to the user.

`.context("some message")?`